sprokit/processes/ocv/detect_in_subregions_process.cxx

```diff
@@ -0,0 +1,377 @@
+
+      for( auto det : *dets )
+      {
+        auto det_bbox = det->bounding_box();
+        vital::bounding_box_d::vector_type offset( 2 );
+        offset[0] = x; offset[1] = y;
+        kwiver::vital::translate( det_bbox, offset );
+        det->set_bounding_box( det_bbox );
+      }
+
+      // Add detections set to the output detection set
+      dets_out.insert( dets_out.end(), dets->cbegin(), dets->cend() );
+
+      timer.stop();
+
+      processed_count++;
+      previous_regions.push_back( roi );
+
+      LOG_TRACE( m_logger, "Time to classify window: " << timer.elapsed() );
+    } // end foreach
+
+    dets_out_sptr = std::make_shared<vital::detected_object_set>( dets_out );
+  }
+
+}; // end priv class
+
+
+// =============================================================================
+detect_in_subregions_process
+::detect_in_subregions_process( kwiver::vital::config_block_sptr const& config )
+  : process( config ),
+    d( new detect_in_subregions_process::priv )
+{
+  // Attach our logger name to process logger
+  attach_logger( kwiver::vital::get_logger( name() ) ); // could use a better approach
+  d->m_logger = logger();
+
+  make_ports();
+  make_config();
+}
+
+
+detect_in_subregions_process
+::~detect_in_subregions_process()
+{
+}
+
+
+//------------------------------------------------------------------------------
+void
+detect_in_subregions_process
+::_configure()
+{
+  scoped_configure_instrumentation();
+
+  vital::config_block_sptr algo_config = get_config();
+
+  // Check config so it will give run-time diagnostic of config problems
+  if( !vital::algo::image_object_detector::
+       check_nested_algo_configuration( "detector", algo_config ) )
+  {
+    throw sprokit::invalid_configuration_exception( name(), "Configuration check failed." );
+  }
+
+  vital::algo::image_object_detector::
+    set_nested_algo_configuration( "detector", algo_config, d->m_detector );
+
+  if( !d->m_detector )
+  {
+    throw sprokit::invalid_configuration_exception( name(), "Unable to create detector" );
+  }
+
+  std::string method = config_value_using_trait( method );
+
+  if( method == "detection_box" )
+  {
+    d->m_method = priv::DETECTION_BOX;
+  }
+  else if( method == "fixed_size" )
+  {
+    d->m_method = priv::FIXED_SIZE;
+  }
+  else
+  {
+    throw sprokit::invalid_configuration_exception( name(), "Invalid method: " + method );
+  }
+
+  d->m_max_subregion_count = config_value_using_trait( max_subregion_count );
+  d->m_fixed_size = config_value_using_trait( fixed_size );
+  d->m_threshold = config_value_using_trait( threshold );
+  d->m_include_input_dets = config_value_using_trait( include_input_dets );
+}
+
+
+// -----------------------------------------------------------------------------
+void
+detect_in_subregions_process
+::_step()
+{
+  vital::detected_object_set_sptr dets_out;
+  double elapsed_time(0);;
+
+  LOG_TRACE( logger(), "Starting process" );
+
+  kwiver::vital::wall_timer timer;
+  timer.start();
```

Comment:
Starting the timer before getting the input data can lead to erroneous timing. It is possible for those grab_* calls to block for a while giving inflated processing times.